### ISSUE TITLE
commit message template

### ISSUE THREAD

**Issue by gyorb** [Ericsson/codechecker#1924]
Using a git commit template message could improve the commit message quality.

```
# TITLE ... (50 characters)


# WHY 
# Explain what is changed and why is this change needed?

# RELATED
# Provide links to any relevant tickets, articles or other resources

# TAGS
# Tags suitable for searching, such as hashtags, keywords, etc.

```
Save  the text as a file at `~/.commit_template`
Configure git to use it: ```git config --local commit.template ~/.commit_template```

For the title:  
 - Use 50 characters maximum.
 - Do not use a sentence-ending period.

For the body:  
- Use 72 characters maximum per line for typical word wrap text.

See a post on [how to write git commit messages](https://chris.beams.io/posts/git-commit/).

Update:
To apply it only for the current repository.

As an extension we could create some commit message rules base on the vue/angular commit message templates.
[vue](https://github.com/vuejs/vue/blob/dev/.github/COMMIT_CONVENTION.md)
[angular](https://github.com/conventional-changelog/conventional-changelog/tree/master/packages/conventional-changelog-angular)

**Comment by whisperity** [comment]
Use `--local` instead of `--global` in the config so that it only applies to the clones of the current repository, not the whole user.

**Comment by csordasmarton** [comment]
This was fixed in #2471.
